### ISSUE TITLE
[$500] Room - User is navigated to Concierge report when creating room

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#37234]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.44-0
**Reproducible in staging?:** y
**Reproducible in production?:** n
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Create a workspace if you don't have any
2. Click on 'FAB' > Start chat > Room tab
3. Enter room name and click on 'Create room' button

## Expected Result:
User should be navigated to the newly created room

## Actual Result:
User is navigated to Concierge report

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/93399543/ed087f9b-916f-46b1-a672-25b2f4bdd3a0

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0146027f79fc978115</li>
        <li>Upwork Job ID: 1762246166149222400</li>
        <li>Last Price Increase: 2024-02-26</li>
    </ul>
</details>

**Comment by kbecciv** [comment]
We think that this bug might be related to #vip-vsb
|CC @quinthar

**Comment by blimpich** [comment]
Looking into this

**Comment by blimpich** [comment]
@dukenv0307 This is a regression though, right? What PR caused this?

**Comment by blimpich** [comment]
Seems like something happened such that the props getting passed down into `ReportScreen` are incorrect now, though I'm having a hard time finding which PR caused this.

**Comment by blimpich** [comment]
[posted in #expensify-open-source about this issue](https://expensify.slack.com/archives/C01GTK53T8Q/p1708994623027729)

**Comment by blimpich** [comment]
> @dukenv0307 This is a regression though, right? What PR caused this?

This again. Something broke the app, and in order to accept your proposal I would want to know why this started happening in the first place.

**Comment by s77rt** [comment]
Timeline after we press the create room button
1. `Policy.addPolicyReport` is called and it sets the optimistic report
2. `Report.openReport` is called and it overrides the optimistic report **<- This should not happen**

The because chain: 
- `Report.openReport` is called in `ReportScreen` `fetchReportIfNeeded` because `isLoadingInitialReportActions` is `true`. 
- `isLoadingInitialReportActions` is `true` because `reportActions` are empty.
- `reportActions` are empty because `withOnyx` subscription returns `undefined` 
- `withOnyx` subscription returns `undefined` due to a recent change https://github.com/Expensify/react-native-onyx/pull/470 I'm not sure why exactly but it seems to contradicts with the `selector` 


Offending PRs:
- https://github.com/Expensify/App/pull/36996
- https://github.com/Expensify/react-native-onyx/pull/470

**Comment by bernhardoj** [comment]
@s77rt is on point. This has the same root cause as this [issue](https://www.github.com/Expensify/App/issues/37247#issuecomment-1965912623)

**Comment by bernhardoj** [comment]
The PR is here https://github.com/Expensify/App/pull/37269

**Comment by blimpich** [comment]
Reverted the react Onyx change by first reverting the react-onyx package and then [upgrading the App's package](https://github.com/Expensify/App/pull/37323) to the latest version. Thank you @s77rt for the explanation!

And I'm sorry @bernhardoj the timing didn't work out with your fix. Its morning over here and by the time I started work we really wanted to get the deploy out, so it makes sense to do this revert given the time / risk involved. It always sucks when making seemingly straightforward changes has these unforeseeable cascading effects.

Closing as https://github.com/Expensify/App/pull/37323 fixes this.
